docs/network/building-kernel.md

```diff
@@ -12,7 +12,104 @@ This build process creates the installer kernel, which is a compressed version o
 The data for the original kernel is stored in separate files called preimages.
 For more information about installer kernels and preimages, see the tutorial [Deploy a Smart Rollup](https://docs.tezos.com/tutorials/smart-rollup) on docs.tezos.com.
 
-The Etherlink build process relies on a Docker image to ensure reproducible builds, so you must have Docker installed.
+## Prerequisites
+
+Before you begin, make sure that you have these prerequisites installed:
+
+- Docker, because the Etherlink build process relies on a Docker image to ensure reproducible builds
+
+- Rust, because of its support for WebAssembly (WASM), the language that Smart Rollups use to communicate.
+
+   To install Rust via the `rustup` command, run this command:
+
+   ```bash
+   curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+   ```
+
+   You can see other ways of installing Rust at https://www.rust-lang.org.
+
+- Clang and LLVM, which are required for compilation to WebAssembly.
+   Version 11 or later of Clang is required.
+   Here are instructions for installing the appropriate tools on different operating systems:
+
+   **MacOS**
+
+   ```bash
+   brew install llvm
+   export CC="$(brew --prefix llvm)/bin/clang"
+   ```
+
+   In some cases for MacOS you may need to update your `PATH` environment variable to include LLVM by running this command:
+
+   ```bash
+   echo 'export PATH="/opt/homebrew/opt/llvm/bin:$PATH"' >> ~/.zshrc
+   ```
+
+   **Ubuntu**
+
+   ```bash
+   sudo apt-get install clang-11
+   export CC=clang-11
+   ```
+
+   **Fedora**
+
+   ```bash
+   dnf install clang
+   export CC=clang
+   ```
+
+   **Arch Linux**
+
+   ```bash
+   pacman -S clang
+   export CC=clang
```

Comment:
I’m not sure this is a strong requirement, at least I’m pretty sure I never had to do it.

Reply:
This comes from instructions I got a while back for the Smart Rollup tutorial. I don't have any way of testing them.